### PR TITLE
[AddressTablePagination] - add component

### PR DESCRIPTION
**Pull request by pietro-maximoff**
https://sovryn.atlassian.net/browse/SOV-680

### CODE REVIEW

File: packages/ui/src/2_molecules/AddressTablePagination/AddressTablePagination.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import classnames from 'classnames';
+
+import { Icon } from '../../1_atoms';
+import styles from './AddressTablePagination.module.css';
+
+type AddressTablePaginationProps = {
+  onPageChange: (value: number) => void;
+  className?: string;
+  dataLayoutId?: string;
+  limit?: number;
```
Comment (tiltom) [inline packages/ui/src/2_molecules/AddressTablePagination/AddressTablePagination.tsx]:
```suggestion
  itemsPerPage?: number;
```

File: packages/ui/src/2_molecules/AddressTablePagination/AddressTablePagination.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+};
+
+const DEFAULT_LIMIT = 5;
```
Comment (tiltom) [inline packages/ui/src/2_molecules/AddressTablePagination/AddressTablePagination.tsx]:
```suggestion
const DEFAULT_ITEMS_PER_PAGE = 5;
```

File: packages/ui/src/2_molecules/AddressTablePagination/AddressTablePagination.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+export const AddressTablePagination: FC<AddressTablePaginationProps> = ({
+  onPageChange,
+  className,
+  dataLayoutId,
+  limit = DEFAULT_LIMIT,
+}) => {
+  const [offset, setOffset] = useState(0);
+
+  const handleMoveRight = useCallback(() => {
```
Comment (tiltom) [inline packages/ui/src/2_molecules/AddressTablePagination/AddressTablePagination.tsx]:
```suggestion
  const handleClickNext = useCallback(() => {
```

File: packages/ui/src/2_molecules/AddressTablePagination/AddressTablePagination.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    setOffset(offset + limit);
+    onPageChange(offset);
+  }, [setOffset, offset, limit, onPageChange]);
+
+  const handleMoveLeft = useCallback(() => {
```
Comment (tiltom) [inline packages/ui/src/2_molecules/AddressTablePagination/AddressTablePagination.tsx]:
```suggestion
  const handleClickPrevious = useCallback(() => {
```

File: packages/ui/src/2_molecules/AddressTablePagination/AddressTablePagination.test.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import React from 'react';
+
+import { AddressTablePagination } from './AddressTablePagination';
+
+describe('AddressTablePagination', () => {
+  it('renders a AddressTablePagination', () => {
+    const onPageChange = jest.fn();
+    render(<AddressTablePagination onPageChange={onPageChange} />);
+    expect(screen.findAllByRole('svg[data-icon="arrow-back"]')).toBeDefined();
+  });
```
Comment (tiltom) [inline packages/ui/src/2_molecules/AddressTablePagination/AddressTablePagination.test.tsx]:
```suggestion
  it('renders AddressTablePagination', () => {
    const onPageChange = jest.fn();
    const { findAllByRole } = render(
      <AddressTablePagination onPageChange={onPageChange} />,
    );
    expect(findAllByRole('svg[data-icon="arrow-back"]')).toBeDefined();
  });
```

File: packages/ui/src/2_molecules/AddressTablePagination/AddressTablePagination.test.tsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+  it('does not allow to click on a disabled button', () => {
```
Comment (tiltom) [inline packages/ui/src/2_molecules/AddressTablePagination/AddressTablePagination.test.tsx]:
```suggestion
  it('does not allow to click Previous without clicking Next first', () => {
```

File: packages/ui/src/2_molecules/AddressTablePagination/AddressTablePagination.test.tsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    const onPageChange = jest.fn();
+    render(<AddressTablePagination onPageChange={onPageChange} />);
```
Comment (tiltom) [inline packages/ui/src/2_molecules/AddressTablePagination/AddressTablePagination.test.tsx]:
```suggestion
    const { getAllByRole } = render(<AddressTablePagination onPageChange={onPageChange} />);
```

File: packages/ui/src/2_molecules/AddressTablePagination/AddressTablePagination.test.tsx [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    screen.getAllByRole('button').forEach(element => {
```
Comment (tiltom) [inline packages/ui/src/2_molecules/AddressTablePagination/AddressTablePagination.test.tsx]:
```suggestion
    getAllByRole('button').forEach(element => {
```

File: packages/ui/src/2_molecules/AddressTablePagination/AddressTablePagination.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    <AddressTablePagination {...args} />
+  </>
+);
+
+export const Basic = Template.bind({});
+Basic.args = {
+  className: '',
+  dataLayoutId: '',
+  limit: 5,
+  onPageChange: noop,
```
Comment (tiltom) [inline packages/ui/src/2_molecules/AddressTablePagination/AddressTablePagination.stories.tsx]: I'd add here some `alert` or `console.log` to indicate what's happening (e.g. `console.log` with information about what items we should see).

**Review by tiltom — CHANGES_REQUESTED** [review]

**Review by soulBit — COMMENTED** [review]
Reviewed, no additional comments than the ones tiltom left

File: packages/ui/src/2_molecules/AddressTablePagination/AddressTablePagination.tsx [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+const DEFAULT_ITEMS_PER_PAGE = 5;
+
+export const AddressTablePagination: FC<AddressTablePaginationProps> = ({
+  onPageChange,
+  className,
+  dataLayoutId,
+  itemsPerPage = DEFAULT_ITEMS_PER_PAGE,
+}) => {
+  const [offset, setOffset] = useState(0);
+  console.log(offset);
```
Comment (creed-victor) [inline packages/ui/src/2_molecules/AddressTablePagination/AddressTablePagination.tsx]:
```suggestion
```

**Review by creed-victor — CHANGES_REQUESTED** [review]

**Review by creed-victor — APPROVED** [review]

**Review by Rickk137 — APPROVED** [review]

**Review by tiltom — APPROVED** [review]

**Pull request merged**